crates/wasmparser/src/validator/operators.rs

```diff
@@ -560,6 +592,20 @@ impl<'resources, R: WasmModuleResources> OperatorValidatorTemp<'_, 'resources, R
         Ok(())
     }
 
+    fn check_function_references_enabled(&self, offset: usize) -> Result<()> {
+        if !self.features.function_references {
+            bail!(offset, "function references support is not enabled",);
+        }
+        Ok(())
+    }
+
+    fn check_tail_call_enabled(&self, offset: usize) -> Result<()> {
+        if !self.features.tail_call {
+            bail!(offset, "tail call support is not enabled",);
+        }
+        Ok(())
+    }
```

Comment:
With recent refactorings it's ok to remove these  since it's handled by the `validate_proposal` macro